src/system/exceptions.hh

```diff
@@ -321,6 +322,37 @@ TYPEDEF_ERR_INFO( EI_XprintfMessage, std::string);
 DECLARE_EXCEPTION( ExcXprintfMsg, << EI_XprintfHeader::val << EI_XprintfMessage::val);
 
 
+/**
+ * @brief Error info of previous exception.
+ *
+ * Allows keep and propagate message when one exception is catched and other exception is thrown.
+ * Catched exception is stored to EI_Nested and is printed out to message of thrown exception.
+ *
+ * Example of usage:
+ *
+ @code
+	try {
+		// method next() throws ExcA
+		obj.next();
+	} catch ( ExcA &e ) {
+	    // add ExcA to EI tags of ExcB
+		THROW( ExcB() << EI_Nested(make_exception_ei<ExcA>(e)) );
+	}
+ @endcode
+ *
+ */
+TYPEDEF_ERR_INFO( EI_Nested, std::shared_ptr<ExceptionBase>);
+
+
+/**
+ * Create shared pointer of given exception.
+ *
+ * Used for propagation exception message.
+ */
+template <class Exc>
+std::shared_ptr<Exc> make_exception_ei(Exc &e) {
+    return std::make_shared<Exc>(e);
```

Comment:
check than Exc is derived from ExceptionBase using 
[type traits](http://en.cppreference.com/w/cpp/header/type_traits)
in combination with STATIC_ASSERT